File: trunk/PYTHON/pyzbus/test/loadbalance/client/AsynSend.py
from pyzbus import BusClient, AsynCtrl
import time

client = BusClient(host='127.0.0.1', port=15555)

ctrl = AsynCtrl()
ctrl.service = 'MyService' 
ctrl.msg_id = int(time.time())
ctrl.peer_id = 'local_mq'
ctrl.timeout = 2500 #ms

message = ['hello @%s'%time.time()]

result = client.send(ctrl, message) 
print result

client.destroy()    

File: trunk/MS.NET/clrzbus/examples/loadbalance/client/AsynSend.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using zbus;
namespace zbus
{
    class AsynSend
    {
        public static void Main(string[] args)
        {
            //1) 创建连接
            ConnectionConfig config = new ConnectionConfig();
            config.Host = "127.0.0.1";
            config.Port = 15555;  
            BusClient client = new BusClient(config);

            //2) 异步控制结构
            AsynCtrl ctrl = new AsynCtrl();
            ctrl.Service = "MyService";
            ctrl.PeerId = "local_mq";//消息投递目标ID，与接收方协商
            ctrl.MessageId = "10000"; 
            ctrl.Timeout = 2500; ////异步投递失败提示最长时间(ms)


            //3) 消息体（帧组成）
            ZMsg msg = new ZMsg();
            msg.PushBack("asyn call frame1");
            msg.PushBack("asyn call frame2");

            //4) 发送异步消息
            msg = client.Send(ctrl, msg);
            msg.Dump();

            //5）销毁链接
            client.Destroy();

            Console.ReadKey();
        } 
    }
}
